### PR TITLE
Fix for multiple wait activities on Android (https://github.com/appium/appium-adb/issues/138)

### PR DESCRIPTION
**Pull request by amaharana**

### CODE REVIEW

File: lib/tools/apk-utils.js [hunk 1]
```diff
@@ -117,8 +117,15 @@ apkUtilsMethods.waitForActivityOrNot = async function (pkg, activity, not,
...
             `${not ? ' not' : ''} to be focused`);
   let endAt = Date.now() + waitMs;
-  let possibleActivityNames = getPossibleActivityNames(pkg, activity);
+  
+  let possibleActivityNames = [];
+  let allActivities = activity.split(",");
+  allActivities.forEach(function (oneActivity) {
+    possibleActivityNames.push(getPossibleActivityNames(pkg, oneActivity.trim()));
+  });
+  possibleActivityNames = [].concat.apply([], possibleActivityNames);
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: What is this line doing, other than cloning `possibleActivityNames`? I'm not sure it is necessary.

File: lib/tools/apk-utils.js [hunk 2]
```diff
@@ -117,8 +117,15 @@ apkUtilsMethods.waitForActivityOrNot = async function (pkg, activity, not,
   log.debug(`Waiting for pkg: '${pkg}' and activity: '${activity}'` +
             `${not ? ' not' : ''} to be focused`);
   let endAt = Date.now() + waitMs;
-  let possibleActivityNames = getPossibleActivityNames(pkg, activity);
+  
+  let possibleActivityNames = [];
+  let allActivities = activity.split(",");
+  allActivities.forEach(function (oneActivity) {
```
Comment (imurchie) [inline lib/tools/apk-utils.js]: The Appium style is to use ES2015 `for of` loops:

```js
for (let activity of allActivities) {
  activity = activity.trim();
  possibleActivityNames.push(...getPossibleActivityNames(pkg, activity));
}
```

**Comment by imurchie** [comment]
Looks good! Thank you for this!

**Comment by imurchie** [comment]
There is a need, however, for some tests. I've make a few that exercise this feature. They, or something like them, ought to be added to https://github.com/appium/appium-adb/blob/master/test/unit/apk-utils-specs.js#L121

```js
    it('should be able to get first of a comma-separated list of activities', async () => {
      mocks.adb.expects('shell')
        .once().withExactArgs(['dumpsys', 'window', 'windows'])
        .returns(`mFocusedApp=AppWindowToken{38600b56 token=Token{9ea1171 ` +
                 `ActivityRecord{2 u ${pkg}/.ContactManager t181}}}`);

      await adb.waitForActivityOrNot(pkg, '.ContactManager, .OtherManager', false);
      mocks.adb.verify();
    });
    it('should be able to get second of a comma-separated list of activities', async () => {
      mocks.adb.expects('shell')
        .once().withExactArgs(['dumpsys', 'window', 'windows'])
        .returns(`mFocusedApp=AppWindowToken{38600b56 token=Token{9ea1171 ` +
                 `ActivityRecord{2 u ${pkg}/.OtherManager t181}}}`);

      await adb.waitForActivityOrNot(pkg, '.ContactManager, .OtherManager', false);
      mocks.adb.verify();
    });
    it('should fail if no activity in a comma-separated list is available', async () => {
      mocks.adb.expects('shell')
        .atLeast(1)
        .withExactArgs(['dumpsys', 'window', 'windows'])
        .returns(`mFocusedApp=AppWindowToken{38600b56 token=Token{9ea1171 ` +
                 `ActivityRecord{2 u ${pkg}/${act} t181}}}`);

      await adb.waitForActivityOrNot(pkg, '.SuperManager, .OtherManager', false, 1000)
        .should.eventually.be.rejected;
      mocks.adb.verify();
    });
```

File: lib/tools/apk-utils.js (hunk 1 above)
Comment (amaharana) [inline lib/tools/apk-utils.js]: The return from getPossibleActivityNames() is an array. So the elements in possibleActivityNames are arrays and they will fail the _.findIndex() further down unless possibleActivityNames is flattened which is what this line does.

Comment (imurchie) [inline lib/tools/apk-utils.js]: That makes sense. In ES2015 there is the `...` operator which "spreads" an array into individual elements. See the above snippet I suggested for the `for` loop.

Comment (amaharana) [inline lib/tools/apk-utils.js]: Perfect. Will update the pull request once I have the changes approved internally.

**Comment by imurchie** [comment]
Travis refuses to start jobs.

Can you squash these two commits into one? I will then test locally. I can't guarantee that it will get in the next version of Appium since it involves a lot of testing that I don't really have time to redo.

**Comment by amaharana** [comment]
Done 6ed4c37

**Comment by amaharana** [comment]
The Travis CI build is stuck ... anything I can do to help? Kill this build and start a fresh one maybe?

**Pull request closed without merging**